.github/workflows/tests.yml

```diff
@@ -5,6 +5,7 @@ on: [push]
 jobs:
   ubuntu:
     runs-on: ubuntu-latest
+    timeout-minutes: 30
```

Comment:
do we still want to keep this? Could be even lower I guess

Reply:
It's just a safety guard from spending too many github action credits. I think I want to keep it.